src/modules/notifications/notifications.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable, InternalServerErrorException, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Notification } from './entities/notifications.entity';
+import { User } from '../user/entities/user.entity';
+
+@Injectable()
+export class NotificationsService {
+  constructor(
+    @InjectRepository(Notification)
+    private notificationRepository: Repository<Notification>,
+    @InjectRepository(User)
+    private userRepository: Repository<User>
+  ) {}
+
+  async getUnreadNotificationsForUser(userId: string) {
+    try {
+      const user = await this.userRepository.findOne({ where: { id: userId } });
+      if (!user) {
+        throw new NotFoundException('User not found');
+      }
+
```

Comment:
All these logic already exists in the userService.

```suggestion
const user: Partial<UserInterface> = await this.userService.getUserRecord({
      identifier: user_id,
      identifierType: 'id',
    });
```

if no user it is a Bad request not 404... Also test your error responses conform with the issue description, because I can guess this will not

Reply:
resolved

Reply:
You haven't resolved. You're still querying the user dB when you don't need to